src/Main.cpp

```diff
@@ -1,10 +1,467 @@
+            }
+            else
+            {
+                return;
+            }
+
+            // Sdp negotiation parser
+            if (rusty_websocket)
+            {
+                if (actiontype == "offer" || actiontype == "answer")
+                {
+                    offerAnswerSignalMessageParser(rusty_websocket, signalr_decoder);
+                }
+                else if (actiontype == "candidate")
+                {
+                    candidateSignalMessageParser(rusty_websocket, signalr_decoder);
+                }
+            }
+        }
+    );
+
+    // wss://localhost:5001/sessionHub?id=+connectionToken
+    const string signalr_url_websocket =
+        local_peer_id + ":5001/sessionHub?id=" + https_token;
+    cout << "SignalR WebSocket URL is " << signalr_url_websocket << endl;
+    signalr_websocket->open(signalr_url_websocket);
+    sendInitialPayload(signalr_websocket);
+
+    RustySignalMessageDecoder rusty_decoder = RustySignalMessageDecoder();
+    promise<void> ws_promise;
+    future<void> ws_future = ws_promise.get_future();
+    rtc::Configuration rusty_config;
+    // TODO - make it configurable
+    rusty_config.iceServers.emplace_back("stun:stun.l.google.com:19302");
+
+    rusty_websocket->onOpen(
+        [&]()
+        {
+            cout << "RustySignal WebSocket connected, signaling ready" << endl;
+            ws_promise.set_value();
+        }
+    );
+
+    rusty_websocket->onError(
+        [&](string const& error)
+        {
+            cout << "RustySignal WebSocket failed: " << error << endl;
+            ws_promise.set_exception(make_exception_ptr(runtime_error(error)));
+        }
+    );
+
+    rusty_websocket->onClosed([&]() { cout << "RustySignal WebSocket closed" << endl; });
+
+    rusty_websocket->onMessage(
+        [&](variant<binary, string> data)
+        {
+            if (!holds_alternative<string>(data))
+            {
+                return;
+            }
+
+            string error;
+            if (!rusty_decoder.parseJSONMessage(get<string>(data).c_str(), error))
+            {
+                throw invalid_argument(error);
+            }
+
+            string actiontype = rusty_decoder.getActionType();
+
+            if (actiontype == "ping" && client_id_checked)
+            {
+                pong(rusty_websocket, rusty_decoder, local_peer_id);
+            }
+
+            if (signalr_websocket)
+            {
+                if (actiontype == "offer" || actiontype == "answer")
+                {
+                    offerAnswerRustyMessageParser(signalr_websocket, rusty_decoder);
+                }
+                else if (actiontype == "candidate")
+                {
+                    candidateRustyMessageParser(signalr_websocket, rusty_decoder);
+                }
+            }
+        }
+    );
+
+    // wss://signalserverhost?user=yourname
+    const string rusty_url_websocket = "127.0.0.1:3012?user=" + local_peer_id;
```

Comment:
Same. This needs to be configurable. Maybe actually make them configurable right now ? As arguments to the binary...